microsoft/testsuites/network/synthetic.py

```diff
@@ -86,6 +87,7 @@ def synthetic_add_max_nics_one_time_after_provision_validation(
         3. Check each nic has an ip address.
         """,
         priority=2,
+        use_new_environment=True,
```

Comment:
What's the reason they need new environments?

Reply:
from current test results, it will reuse previous environment which already attached 8 nics, and I found use_new_environment seems not working...

Reply:
also for any provision test case, we should use new environment?

Reply:
Every test case should recover the environment after it's completed. It can be done in `after_case`. It runs even the test case failed earlier. Let me know what's the behavior of "not working".

Reply:
synthetic_provision_with_max_nics_validation
synthetic_add_max_nics_one_time_after_provision_validation
synthetic_add_max_nics_one_by_one_after_provision_validation
will run in sequence, the first case will provision with 8 nics, then the second case will reuse previous environment, at that time, the VM is not allowed to add any nic any more. case will fail during attach nic.
![image](https://user-images.githubusercontent.com/10083705/144396362-1e1f9dd8-8a5d-4fbb-9ef1-b309fdea273b.png)


should we remove the nics after run case synthetic_provision_with_max_nics_validation?

Reply:
@squirrelsc thoughts?

Reply:
1. The logic is needed in `after_case` to clean up all added nics. It makes sure the environment is recovered, and ready for next test case.
2. I mean what's the behavior of the `use_new_environment`  not working, from your comments? "I found use_new_environment seems not working..." I had a try, the use_new_environment works on my scenario.

Reply:
synthetic_provision_with_max_nics_validation
synthetic_add_max_nics_one_time_after_provision_validation + use_new_environment 
synthetic_add_max_nics_one_by_one_after_provision_validation + use_new_environment 

I saw below two cases not creating new environment.

Reply:
Can you share me a log?

Reply:
logs sent to you by mail, 
update code to remove extra nic in after_case